evalml/pipelines/time_series_classifcation_pipelines.py

```diff
@@ -0,0 +1,204 @@
+
+    def __init__(self, parameters, random_state=0):
+        """Machine learning pipeline for time series regression problems made out of transformers and a classifier.
+
+        Required Class Variables:
+            component_graph (list): List of components in order. Accepts strings or ComponentBase subclasses in the list
+
+        Arguments:
+            parameters (dict): Dictionary with component names as keys and dictionary of that component's parameters as values.
+                 An empty dictionary {} implies using all default values for component parameters. Pipeline-level
+                 parameters such as gap and max_delay must be specified with the "pipeline" key. For example:
+                 Pipeline(parameters={"pipeline": {"max_delay": 4, "gap": 2}}).
+            random_state (int, np.random.RandomState): The random seed/state. Defaults to 0.
+        """
+        if "pipeline" not in parameters:
+            raise ValueError("gap and max_delay parameters cannot be omitted from the parameters dict. "
+                             "Please specify them as a dictionary with the key 'pipeline'.")
+        pipeline_params = parameters["pipeline"]
+        self.gap = pipeline_params['gap']
+        self.max_delay = pipeline_params['max_delay']
+        super().__init__(parameters, random_state)
+
+    @staticmethod
+    def _convert_to_woodwork(X, y):
+        if X is None:
+            X = pd.DataFrame()
+        X = _convert_to_woodwork_structure(X)
+        y = _convert_to_woodwork_structure(y)
+        return X, y
+
+    def fit(self, X, y):
+        """Fit a time series regression pipeline.
+
+        Arguments:
+            X (ww.DataTable, pd.DataFrame or np.ndarray): The input training data of shape [n_samples, n_features]
+            y (ww.DataColumn, pd.Series, np.ndarray): The target training targets of length [n_samples]
+
+        Returns:
+            self
+        """
+        X, y = self._convert_to_woodwork(X, y)
+        X = _convert_woodwork_types_wrapper(X.to_dataframe())
+        y = _convert_woodwork_types_wrapper(y.to_series())
+        self._encoder.fit(y)
+        y = self._encode_targets(y)
+
+        X_t = self._compute_features_during_fit(X, y)
+        if X_t.empty:
+            raise RuntimeError("Pipeline computed empty features during call to .fit. This means "
+                               "that either 1) you passed in X=None to fit and don't have a DelayFeatureTransformer "
+                               "in your pipeline or 2) you do have a DelayFeatureTransformer but gap=0 and max_delay=0. "
+                               "Please add a DelayFeatureTransformer or change the values of gap and max_delay")
+
+        y_shifted = y.shift(-self.gap)
+        X_t, y_shifted = drop_rows_with_nans(X_t, y_shifted)
+        self.estimator.fit(X_t, y_shifted)
+        return self
+
+    def _predict(self, X, y, objective=None, pad=False):
```

Comment:
I have added a `pad` parameter to `_predict` because in `predict` we want to return un-padded predictions or else `self._decode_targets` will error out. However in `_compute_predictions` (called by `score`) we want to return padded predictions (so that predictions, predicted probabilities, and the shifted target all have the same length. This makes dropping nans much easier).